gui/src/app/Stanc/useStanc.ts

```diff
@@ -0,0 +1,67 @@
+import { useCallback, useEffect, useState } from "react";
+import { StancReplyMessage, StancErrors, Replies, Requests } from "./Types";
+import stancWorkerURL from "./stancWorker?worker&url";
+
+const useStanc = (
+  modelName: string,
+  code: string,
+  onFormat: (s: string) => void,
+) => {
+  const [stancErrors, setStancErrors] = useState<StancErrors>({});
+  const [stancWorker, setStancWorker] = useState<Worker | undefined>(undefined);
+
+  // worker creation
+  useEffect(() => {
+    const worker = new Worker(stancWorkerURL, {
+      name: "stancWorker",
+      type: "module",
+    });
+    setStancWorker(worker);
+    return () => {
+      console.log("terminating stanc worker");
+      worker.terminate();
+    };
+  }, []);
+
+  // message handling
+  useEffect(() => {
+    if (!stancWorker) return;
+
+    stancWorker.onmessage = (e: MessageEvent<StancReplyMessage>) => {
+      const { purpose, result, error } = e.data;
+      if (error) {
+        // not loaded yet
+        console.error(error);
+        return;
+      }
+      setStancErrors({ ...e.data });
+      if (purpose === Replies.Formatted) {
+        if (result) {
+          onFormat(result);
+        }
+      }
+    };
```

Comment:
Also a place where just a smidge more documentation of the API will go a long way to me understanding this in two weeks when I've forgotten what it does :sweat_smile: